packages/wouter/src/use-hash-location.js

```diff
@@ -22,11 +22,31 @@ const subscribeToHashUpdates = (callback) => {
 // leading '#' is ignored, leading '/' is optional
 const currentHashLocation = () => "/" + location.hash.replace(/^#?\/?/, "");
 
-export const navigate = (to, { state = null } = {}) => {
+export const navigate = (to, { state = null, replace = false } = {}) => {
   // calling `replaceState` allows us to set the history
   // state without creating an extra entry
   const [hash, search] = to.replace(/^#?\/?/, "").split("?");
 
+  if (replace) {
+    history.replaceState(
+      state,
+      "",
+      // keep the current pathname, but replace query string and hash
+      location.pathname +
+        (search ? `?${search}` : location.search) +
+        `#/${hash}`
+    );
+
+    const event =
+      typeof HashChangeEvent !== "undefined"
+        ? new HashChangeEvent("hashchange")
+        : new Event("hashchange");
```

Comment:
I can see that this event has also `newURL` and `oldURL` properties, maybe we should add them to the options?

Reply:
yeah that's a good call 👍